### DISCUSSION TITLE
有没有 dde-file-manager 扩展或插件的接口？

### DISCUSSION THREAD

**Discussion by wan2004** [linuxdeepin/developer-center#3538] [Q&A | 问答板块]
请问下： 有没有 dde-file-manager 扩展或插件的接口？
类似Qtplugins 加载SO动态库插件的方式扩展的接口有么？
这边想要扩展文件系统，需要针对UOS开发
感谢任何建议和支持！

**Comment by BLumia** [comment]
有，但好像目前没有公开的文档，项目仓库下的 docs 目录下有一些文档但可能略微过时，所以实际写插件的话可能需要顺着源码摸过去，比如可以参照 ~~`dde-file-manager-plugins`~~ `dde-file-manager-extensions` 的用法。

如果你是厂商的话，建议以厂商的形式联系下看看有没有面向厂商的文档。如果是社区贡献者的话可以考虑加 <https://matrix.to/#/#deepin-community:matrix.org> 一起讨论。

**Reply by wan2004** [reply]
看了一个月代码  dde-file-manager 5.8.3 以及之前的版本 dde-file-manager-plugins 里面接口都没有调用。参照 dde-file-manager-extensions 里面接口做了菜单插件，才行的。

**Reply by BLumia** [reply]
可能 codebase 变化还挺大，完全没用到的话应该删掉的。上次看文管代码比较久远了，那时候还没有 dde-file-manager-extensions，抱歉如果这个答案一定程度上给你造成了误导 :joy:
